gemini/sample-apps/gemini-mesop-cloudrun/main.py

```diff
@@ -0,0 +1,1650 @@
+    marketing_campaign_output: str = ""
+
+    # Image playground
+    image_tab: str = "furniture"
+
+    furniture_recommendation_output: str = ""
+    oven_instructions_output: str = ""
+    er_doc_output: str = ""
+    image_glasses_shape_radio_value: str = "oval"
+    image_glasses_output_radio_value: str = "text"
+    glasses_rec_output: str = ""
+    math_answers_output: str = ""
+
+    image_progress_spinner: bool = False
+
+    # Video playground
+    video_tab: str = "desc"
+
+    video_spinner_progress: bool = False
+    video_description_content: str = ""
+    video_tags_content: str = ""
+    video_highlights_content: str = ""
+    video_geolocation_content: str = ""
+
+
+# Helpers
+def gcsurl_to_httpurl(gcsuri: str) -> str:
+    """given a GCS URI, return the HTTPS URL
+
+    Args:
+        gcsuri (str): Google Cloud Storage URI
+
+    Returns:
+        string: the HTTPS URL equivalent
+    """
+    return "https://storage.googleapis.com/" + gcsuri.split("gs://")[1]
+
+# Events
+
+def on_input(e: me.InputEvent) -> None:
+    print(f"{e}")
+    state = me.state(State)
+    setattr(state, e.key, e.value)
+
+
+## Story events
+def on_selection_change(e: me.SelectSelectionChangeEvent) -> None:
+    s = me.state(State)
+    s.story_selected_premises = e.values
+    print(f"selected: {s.story_selected_premises}")
+
+
+def on_click_clear_story(e: me.ClickEvent) -> None:
+    """Click event for clearing story text."""
+    state = me.state(State)
+    state.story_output = 0
+
+
+def on_radio_change(event: me.RadioChangeEvent) -> None:
+    s = me.state(State)
+    s.radio_value = event.value
+
+
+def on_length_radio_change(event: me.RadioChangeEvent) -> None:
+    s = me.state(State)
+    s.story_length_value = event.value
+
+
+def generate_story(e: me.ClickEvent | me.EnterEvent) -> None:
+    s = me.state(State)
+    s.story_output = "" # clear any existing story
+    s.story_progress = True
+    yield
+
+    if s.story_temp_value == "low":
+        temp = 0.5
+    elif s.story_temp_value == "high":
+        temp = 1.5
+    else:
+        temp = 1
+    prompt = f"""Write a {s.story_length_value} story based on the following premise: \n
+    character_name: {s.story_character_name} \n
+    character_type: {s.story_character_type} \n
+    character_persona: {s.story_character_personality} \n
+    character_location: {s.story_character_location} \n
+    story_premise: {",".join(s.story_selected_premises)} \n
+    If the story is "short", then make sure to have 5 chapters or else if it is "long" then 10 chapters.
+    Important point is that each chapters should be generated based on the premise given above.
+    First start by giving the book introduction, chapter introductions and then each chapter. It should also have a proper ending.
+    The book should have prologue and epilogue.
+    """
+    print(f"prompt: {prompt}")
+
+    model = GenerativeModel(s.model)
+    config = GenerationConfig(temperature=temp, max_output_tokens=2048)
```

Comment:
Why is the max output tokens set low?

Reply:
This was to align with the Streamlit example